packages/component-base/src/polylit-mixin.js

```diff
@@ -0,0 +1,267 @@
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
+
+const caseMap = {};
+
+const CAMEL_TO_DASH = /([A-Z])/g;
+
+function camelToDash(camel) {
+  return caseMap[camel] || (caseMap[camel] = camel.replace(CAMEL_TO_DASH, '-$1').toLowerCase());
+}
+
+function upper(name) {
+  return name[0].toUpperCase() + name.substring(1);
+}
+
+function parseObserver(observerString) {
+  const [method, rest] = observerString.split('(');
+  const observerProps = rest
+    .replace(')', '')
+    .split(',')
+    .map((prop) => prop.trim());
+
+  return {
+    method,
+    observerProps
+  };
+}
+
+function getOrCreateMap(obj, name) {
+  if (!obj[name]) {
+    obj[name] = new Map();
+    // eslint-disable-next-line no-prototype-builtins
+  } else if (!obj.hasOwnProperty(name)) {
+    // clone any existing entries (superclasses)
+    const map = obj[name];
+    obj[name] = new Map();
+    map.forEach((v, k) => obj[name].set(k, v));
+  }
+  return obj[name];
+}
+
+const PolylitMixinImplementation = (superclass) => {
+  class PolylitMixinClass extends superclass {
+    static createProperty(name, options) {
+      if (options.reflectToAttribute) {
+        options.reflect = true;
+      }
+
+      super.createProperty(name, options);
+    }
+
+    static getOrCreateMap(name) {
+      return getOrCreateMap(this, name);
+    }
+
+    /*
+     * @protected
+     * @override
+     */
+    static finalize() {
+      super.finalize();
+
+      if (Array.isArray(this.observers)) {
+        const complexObservers = this.getOrCreateMap('__complexObservers');
+
+        this.observers.forEach((observer) => {
+          const { method, observerProps } = parseObserver(observer);
+          complexObservers.set(method, observerProps);
+        });
+      }
+    }
+
+    static addInitializer(initializer) {
+      super.addInitializer((instance) => {
+        // TODO: Why is this check needed?
```

Comment:
This code was added by f6bce976611884055ab0d6144ef4f23649a3865e already with a TODO. Can we remove it?

Reply:
Added a test case for the fix and moved the fix to proper place